### ISSUE TITLE
Can't build on iOS - "Podfile.lock: No such file or directory"

### ISSUE THREAD

**Issue by AndrWeisR** [phonegap/phonegap-plugin-push#2193]
### Expected Behaviour
Build in Xcode is successful.

### Actual Behaviour
Build in XCode 9.2 fails with errors. I opened the Xcode workspace, not the project.

PhaseScriptExecution [CP]\ Check\ Pods\ Manifest.lock /Users/xxxxxx/Library/Developer/Xcode/DerivedData/push-efgaimzblszoyqbwasqbnmeshyyc/Build/Intermediates.noindex/push.build/Debug-iphonesimulator/push.build/Script-33A2F61D97683A39270A1262.sh
    cd /Users/xxxxxx/MobileDev/Push/platforms/ios
    /bin/sh -c /Users/xxxxxx/Library/Developer/Xcode/DerivedData/push-efgaimzblszoyqbwasqbnmeshyyc/Build/Intermediates.noindex/push.build/Debug-iphonesimulator/push.build/Script-33A2F61D97683A39270A1262.sh

diff: /Podfile.lock: No such file or directory
diff: /Manifest.lock: No such file or directory
error: The sandbox is not in sync with the Podfile.lock. Run 'pod install' or update your CocoaPods installation.
 

#### Platform and Version (eg. Android 5.0 or iOS 9.2.1)
iOS 10.13.2
I have installed CocoaPods 1.4.0

#### Cordova CLI version and cordova platform version

    cordova --version   = 8.00
    cordova platform version ios = 4.5.4

#### Plugin version

    cordova plugin version | grep phonegap-plugin-push   = 2.1.3

#### Notes
Seems to be similar to issue #1668 
My Podfile.lock does actually exist.

The .sh file that produces the error above contains:

```
diff "${PODS_PODFILE_DIR_PATH}/Podfile.lock" "${PODS_ROOT}/Manifest.lock" > /dev/null
if [ $? != 0 ] ; then
    # print error to STDERR
    echo "error: The sandbox is not in sync with the Podfile.lock. Run 'pod install' or update your CocoaPods installation." >&2
    exit 1
fi
```

It looks like the PODS_PODFILE_DIR_PATH and PODS_ROOT variables are empty.

**Comment by jaigtz88** [comment]
Variable ${PODS_ROOT} was not set and I do below trick. 

Go to Build Phases -> Check Pods Manifest.lock and replace
        diff "${PODS_ROOT}/../Podfile.lock" "${PODS_ROOT}/
        Manifest.lock" > /dev/null
**to**
         diff "${SRCROOT}/Podfile.lock" "${SRCROOT}/Pods/

**Comment by macdonst** [comment]
@AndrWeisR from the looks of the error message it seems something is wrong with your cocoapods install. Did you try running `pod install` as it suggested?

**Comment by AndrWeisR** [comment]
Yes, I did a `pod install`, but it didn't help. I had to do as @jaigtz88 above, and manually edit the command in the build phase to enter `SRCROOT`.

**Comment by jaigtz88** [comment]
@AndrWeisR after i update cordova and cocopods, i dont have that error anymore

**Comment by anasamanp** [comment]
Remove all the files related to pod from your project folder, Close xcode project and run `pod init` and `open -a Xcode Podfile`. Add your pod files again and run `pod install`. Reopen and run your project.

**Comment by GoriX** [comment]
Hi all people, 

Same problem with high sierra 10.13.2 and Xcode 9.2, I had have all, delete, start from zero, more than 10 times, I have tested all posibles fixes from all posts related with this issue but always the same result, not working!!! 

May be a problem with permissions? I don't know.... I changed PODS_PODFILE_DIR_PATH for real path /Users/...../documents..../ios/. and PODS_ROOT too and system return same error, it is as if system can't find or not have access to this path that is obvious that it exists...

Any help from somebody?

Thanks in advance.

**Comment by mikeashelby** [comment]
I have this same error, but I can't find the lines in manifest.lock referenced above to edit...

Is there any progress on this? Or anything else to try?

**Comment by gersonmontenegro** [comment]
I has had (at least per now), write the complete path to the .lock files in order to build the project, just like a show below:

![check_pods](https://user-images.githubusercontent.com/3643808/43955771-b85f85c2-9cf5-11e8-8115-cfa0871f7396.png)
At least it will work for me while I fix the issue with the path variables.

**Comment by ivancho2802** [comment]
solved for me: XCODE, IOS, CORDOVA y IONIC

1 - in xcode change value of $(SRCROOT) for you root project in xcode las var of config build in user-defined enter image description here

2 - changed var in this clic in the plus of up: enter image description here

nota: root project is: location of my project and location where use pod this my firs question thank for watching

[(https://stackoverflow.com/questions/40095677/how-can-i-add-a-symbol-like-srcroot-to-use-in-an-xcode-projects-build-settin/51971532#51971532)]

**Comment by aagaardmorten** [comment]
Did you miss a "  in the new script line?
from
diff "${SRCROOT}/Podfile.lock" "${SRCROOT}/Pods/

to 
diff "${SRCROOT}/Podfile.lock" "${SRCROOT}/Pods/"

**Comment by MarkAMcCorkle** [comment]
IMO, the better solution is to add a couple build user-defined settings.
Go to your Project Target, change to Build Settings tab, press the '+' symbol, "Add User-Defined Setting".  

```
PODS_PODFILE_DIR_PATH = ${SRCROOT}/.
PODS_ROOT = ${SRCROOT}/Pods
```

**Comment by ConorTheDev** [comment]
@MarkAMcCorkle I'm not working on this project but ahhh thanks so much man! I'm trying to do something with Firebase and I wasn't able to build it!

Lifesaver!

**Comment by borjabp99** [comment]
@MarkAMcCorkle I tried this solution but still have the same error :/ ... This is how I have it setup:
Build Settings : 

![screen shot 2018-12-01 at 23 45 38](https://user-images.githubusercontent.com/22194228/49333678-91e60980-f5c3-11e8-8411-369f508699d6.png)


Build Phases:

![screen shot 2018-12-01 at 23 49 06](https://user-images.githubusercontent.com/22194228/49333706-d7a2d200-f5c3-11e8-9c69-2fac1c903bef.jpg)


Error:

![screen shot 2018-12-01 at 23 54 16](https://user-images.githubusercontent.com/22194228/49333744-8c3cf380-f5c4-11e8-9e8b-8de75fc8cfc7.png)


What should I do now??

**Comment by at-duytran** [comment]
> IMO, the better solution is to add a couple build user-defined settings.
> Go to your Project Target, change to Build Settings tab, press the '+' symbol, "Add User-Defined Setting".
> 
> ```
> PODS_PODFILE_DIR_PATH = ${SRCROOT}/.
> PODS_ROOT = ${SRCROOT}/Pods
> ```

Thank you a lot. It's save my whole day.

**Comment by Rahul-Sagore** [comment]
@borjabp99 It should be key value. 

![Screenshot 2019-04-24 at 12 36 03 PM](https://user-images.githubusercontent.com/3587851/56639232-cef38b00-668d-11e9-8579-8daf4ae6dadc.png)

**Comment by Rahul-Sagore** [comment]
@borjabp99 It should be key value pair like:

![Screenshot 2019-04-24 at 12 36 03 PM](https://user-images.githubusercontent.com/3587851/56640939-c8671280-6691-11e9-8be4-637fa2b5a098.png)

**Comment by zaifz** [comment]
Update your cocoapod using the command `sudo gem install cocoapod` in your terminal. I had the same issue and it seems like the cocoapod version was old.

**Comment by HaoYan1208** [comment]
> Update your cocoapod using the command `sudo gem install cocoapod` in your terminal. I had the same issue and it seems like the cocoapod version was old.

It works for me!!!!!!

**Comment by hellotunmbi** [comment]
> sudo gem install cocoapod

Kindly note it is 'cocoapods' not 'cocoapod'
